Gulpfile.coffee

```diff
@@ -76,10 +79,14 @@ gulp.task 'tdd', (done) ->
   return # Since this is async
 
 
-gulp.task 'dist', -> build(false)
-gulp.task 'watch', -> build(true)
-gulp.task 'serve', ->
-  build(true)
+gulp.task 'dist', ['build']
+gulp.task 'watch', ['build'], ->
+  gulp.watch ['src/**/*.coffee', 'src/**/*.cjsx', 'test/**/*.coffee'], ['build', 'test']
+  gulp.watch 'style/**/{*.less, *.css}', ['styles']
+
+gulp.task 'build', ['styles'], -> build(false)
+
+gulp.task 'serve', ['watch'], ->
```

Comment:
Currently `gulp serve` will rebuild the JS file whenever a change is made (no need to restart `gulp serve` from the command line. Does that still work?